### PR TITLE
Rails 4 will call with just one argument during asset pre-compilation

### PR DESCRIPTION
**Pull request by staugaard**
Rails 4 will call with just one argument during asset pre-compilation

/cc @zendesk/octo @ggrossman

### CODE REVIEW

**Comment by morten** [comment]
+1

**Comment by jespr** [comment]
:+1:

File: lib/asset_host_selection.rb [hunk 1]
```diff
@@ -47,7 +47,9 @@ def initialize(selector)
       @selector = selector
     end
 
-    def call(source, request)
+    def call(source, request = nil)
+      return nil if !request
```
Comment (ggrossman) [inline lib/asset_host_selection.rb]: No grief about not using `return nil unless request`?

Comment (grosser) [inline lib/asset_host_selection.rb]: `return unless request # rails 4 does not pass request in`

**Comment by staugaard** [comment]
wow super old and forgotten PR. Inbox is running with this patch: https://github.com/zendesk/zendesk_inbox/blob/master/config/initializers/fix_asset_host_selection.rb

**Comment by ggrossman** [comment]
Merge this and then ONE LESS PATCH

**Pull request merged**